### PR TITLE
Refactor deprecations, improve code style and drop owncloud support

### LINKED ISSUE #46: Update to recent Nextcloud versions
I forked and updated this app to work with nextcloud v20 and v21. I refactored the usage of the deprecated nextcloud api and leveraged the new server container interface and the abstract storage backend. This means that my fork is now independent of the used storage backend. However, those changes are not compatible with owncloud.

I would be happy to send a PR if you are willing to drop support for owncloud.
Feel free to close this issue if not.

### PR DESCRIPTION
**Pull request by st3iny**
Fixes #46

I took the liberty to fix the indentation in ChecksumController.php. Previously, tabs and spaces were mixed there.

As discussed in the linked issue, my changes will inevitably drop support for Owncloud. However, the usage of the modern Nextcloud API should make this app future proof for quite some time to come.

I strongly suggest to bump the version to 1.x.x after merging this PR.

cc @westberliner

### CODE REVIEW

File: lib/Controller/ChecksumController.php [hunk 1]
```diff
@@ -1,82 +1,132 @@
...
+		$file = $mount->getStorage()->fopen($internalPath, 'rb');
+		if (!$file) {
+			return null;
 		}
-}
 
+		$hash = hash_init($type);
+		hash_update_stream($hash, $file);
+		fclose($file);
+		return hash_final($hash);
```
Comment (westberliner) [inline lib/Controller/ChecksumController.php:115]: Not sure if it is a psr-2 thing. Blank line before return.

File: lib/Controller/ChecksumController.php [hunk 2]
```diff
@@ -1,82 +1,132 @@
...
+	}
+
+	private function checkAlgorithmType(string $type): bool {
+		return in_array($type, $this->getAllowedAlgorithmTypes()) && in_array($type, hash_algos());
+	}
+
+	private function getAllowedAlgorithmTypes(): array {
+		return array(
```
Comment (westberliner) [inline lib/Controller/ChecksumController.php:123]: Short array syntax.

File: lib/Listener/LoadAdditionalScriptsListener.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+namespace OCA\Checksum\Listener;
+
+use OCA\Files\Event\LoadAdditionalScriptsEvent;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\Util;
+
+class LoadAdditionalScriptsListener implements IEventListener {
+	public function handle(Event $event): void {
```
Comment (westberliner) [inline lib/Listener/LoadAdditionalScriptsListener.php:32]: Blank line before.

**Review by westberliner — CHANGES_REQUESTED** [review]
Thx for the contribution. Looks good so far.

I'll checked the [guidelines](https://docs.nextcloud.com/server/10.0/developer_manual/general/codingguidelines.html#coding) and if I am correct you are using tabs instead of spaces. Could you switch that? I found some minor cs stuff - I'll will marked them. If you do not have the time I will look into that in a way lots of these issues are mine anyway :)

I'll try to test it tomorrow and merge/fix it.

**Comment by st3iny** [comment]
> I'll checked the guidelines and if I am correct you are using tabs instead of spaces. Could you switch that? 

@westberliner These guidelines state that one should use tabs for indentation (Quote: `Use tabs to indent`). But I'm fine with spaces too.

Thanks for your feedback. I implemented your suggestions and switched the indentation to spaces.

**Comment by westberliner** [comment]
> @westberliner These guidelines state that one should use tabs for indentation (Quote: `Use tabs to indent`). But I'm fine with spaces too.

In the next line:

> A tab is 4 spaces wide.

But I think I misunderstood it. Most of the nextcloud code is tabs. Sorry about that. I will check your changes later that day. Thx so far.

**Comment by westberliner** [comment]
Works :) Release is on the way.

**Pull request merged**